engine/reformulation/core/src/main/java/it/unibz/inf/ontop/answering/resultset/SimpleGraphResultSet.java

```diff
@@ -1,17 +1,19 @@
 package it.unibz.inf.ontop.answering.resultset;
 
-import it.unibz.inf.ontop.exception.OntopConnectionException;
 import it.unibz.inf.ontop.exception.OntopResultConversionException;
 import it.unibz.inf.ontop.spec.ontology.RDFFact;
 
+import it.unibz.inf.ontop.exception.OntopConnectionException;
+
 public interface SimpleGraphResultSet extends GraphResultSet<OntopResultConversionException> {
 
     int getFetchSize() throws OntopConnectionException;
 
     /**
      * TODO: remove this hack
      */
-	@Deprecated
-    void addNewResult(RDFFact assertion);
+    @Deprecated
+    void addNewResult(RDFFact statement);
 
+    void addStatementClosable(AutoCloseable sqlStatement);
```

Comment:
From what I have seen the statement is not a SQL statement but an `OntopStatement` (misleading name).

What about removing this method from the interface, and passing the `OntopStatement` as an argument when building the `SimpleGraphResultSet`?

Reply:
I think this might also be passed as a java.sql Statement to the DefaultSimpleGraphResultSet from inside the SqlQuestStatement where the sqlStatement is held